object/provider.go

```diff
@@ -261,21 +265,32 @@ func UpdateProvider(id string, provider *Provider) (bool, error) {
 
 	affected, err := session.Update(provider)
 	if err != nil {
+		logger.Error(ctx, "failed to update provider", "provider", provider.GetId())
 		return false, err
+	} else {
+		if !CompareRoleMappingSlices(provider.RoleMappingItems, p.RoleMappingItems) {
+			logger.Info(ctx, "successfully updated provider mappings", "provider", provider.GetId())
+		}
+		logger.Info(ctx, "successfully updated provider", "provider", provider.GetId())
 	}
 
 	return affected != 0, nil
 }
 
 func AddProvider(provider *Provider) (bool, error) {
+	ctx := context.Background()
+
 	if provider.Type == "Tencent Cloud COS" {
 		provider.Endpoint = util.GetEndPoint(provider.Endpoint)
 		provider.IntranetEndpoint = util.GetEndPoint(provider.IntranetEndpoint)
 	}
 
 	affected, err := ormer.Engine.Insert(provider)
 	if err != nil {
+		logger.Error(ctx, "failed to add provider", "provider", provider.GetId())
 		return false, err
+	} else {
```

Comment:
You dont really need this "else" operator.
The code is equivalent to this clearer version using "return early" pattern
```
if err != nil {
    logger.Error(ctx, "failed to add provider", "provider", provider.GetId())
    return false, err
}

logger.Info(ctx, "successfully added provider", "provider", provider.GetId())
return affected != 0, nil
```

Reply:
done